Document limitations of weights_only in SECURITY.md and torch.load doc

mikaylagawarecki opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #165645




cc @albanD

mikaylagawarecki committed:
Document limitations of weights_only in SECURITY.md and torch.load doc

[ghstack-poisoned]

mikaylagawarecki committed:
Update on "Document limitations of weights_only in SECURITY.md and torch.load doc"




[ghstack-poisoned]

albanD commented on torch/serialization.py:
```diff
@@ -1296,6 +1296,19 @@ def load(
+        uses ``pickle`` module implicitly, which is known to be insecure.
+        It is possible to construct malicious pickle data which will execute arbitrary code
+        during unpickling. Never load data that could have come from an untrusted
+        source in an unsafe mode, or that could have been tampered with. **Only load data you trust**.
```
```suggestion
        source, or that could have been tampered with. **Only load data you trust**.
```

mikaylagawarecki committed:
Update on "Document limitations of weights_only in SECURITY.md and torch.load doc"


<img width="465" height="262" alt="Screenshot 2025-10-16 at 3 25 59 PM" src="https://github.com/user-attachments/assets/154c1da8-6959-4618-bf0c-c36df104ea92" />



cc albanD

[ghstack-poisoned]

mikaylagawarecki committed:
Update on "Document limitations of weights_only in SECURITY.md and torch.load doc"


<img width="465" height="262" alt="Screenshot 2025-10-16 at 3 25 59 PM" src="https://github.com/user-attachments/assets/154c1da8-6959-4618-bf0c-c36df104ea92" />



cc albanD

[ghstack-poisoned]

albanD reviewed (approved):
One small nit, but SGTM !

mikaylagawarecki committed:
Update on "Document limitations of weights_only in SECURITY.md and torch.load doc"





cc albanD

[ghstack-poisoned]
